src/StoreApi/Utilities/ProductQuery.php

```diff
@@ -95,12 +95,16 @@ public function prepare_objects_query( $request ) {
 			'and'    => 'AND',
 		];
 
+		$all_product_taxonomies = get_taxonomies( array( 'object_type' => array( 'product' ) ), 'names' );
+
 		// Map between taxonomy name and arg key.
-		$taxonomies = [
+		$default_taxonomies = [
```

Comment:
Don't these `$default_taxonomies` also get returned by `get_taxonomies()`? If so, is there any need for `$default_taxonomies` and `$taxonomies = array_merge( $default_taxonomies, $all_product_taxonomies );`

Reply:
Good question @tjcafferkey. `get_taxonomies( array( 'object_type' => array( 'product' ) ), 'names' );` returns all product taxonomies including `product_cat` and `product_tag`. The thing is that the endpoint currently uses `category` instead of `product_cat` and `tag` instead of `product_tag` for parameters. So I have opted to merging with `$default_taxonomies` for backwards compatibility. Would be preferable updating the docs and all code references (if any)?

Reply:
>The thing is that the endpoint currently uses category instead of product_cat and tag instead of product_tag for parameters.

@cpapazoglou it looks like after the merge it uses `product_cat` and `product_tag` unless I am wrong?

<img width="598" alt="Screenshot 2022-04-01 at 11 32 28" src="https://user-images.githubusercontent.com/8639742/161246917-187f7bf8-6755-4d5c-8e6b-c5a5913a4000.png">

Reply:
🤦, sorry for this. I was merging with wrong order. Fixed here https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6152/commits/6885fd266ed8bdefeba8931158475d016af1ba19

Works in my local machine 😁
<img width="568" alt="CleanShot 2022-04-01 at 14 10 03@2x" src="https://user-images.githubusercontent.com/12430020/161252257-36cf03c5-d529-4de3-8c73-16f0e10e9b9e.png">

Reply:
@tjcafferkey is the following preferable?
```
$all_product_taxonomies = get_taxonomies( array( 'object_type' => array( 'product' ) ), 'names' );
$all_product_taxonomies['product_cat'] = 'category';
$all_product_taxonomies['product_tag'] = 'tag';
```

Reply:
Thanks @cpapazoglou this seems to work as expected now. However like @Aljullu mentioned I will leave the final review for @opr

Reply:
> is the following preferable?

I think as it currently is works fine.